docs/DeveloperGuide.md

```diff
@@ -1,38 +1,308 @@
+and display the appropriate acknowledgements or required information based on the user's command.
+
+The following diagram illustrates the methods within the Ui class that can be invoked by the other classes in Duke for the purpose of user interaction.
+
+![Ui Class Diagram](./images/Ui_Class.png)
+
+### 1.6 Commands
+
+To interact with easySEP, users have to input commands specified with parameters to perform operations which will be passed to the CommandParser to generate a corresponding Command according to their input.
+The Command will then be executed to perform the operation on the other classes managing the timetables and databases. Error checking is handled to throw InvalidUserCommandException if the user's input does not match
+the specified Command format. To deal with parsing parameters, spaces in University names and Module codes are to be replaced with underscores.
+
+The following class diagram illustrates the relationship between Command class and its subclasses as well as other classes related to Commands.
+![Command Class Diagram](./images/Command_Class.png)
+
+#### 1.6.1 Create Command
+
+A Create command can be used to create a university list and its corresponding timetable.
+
+The following sequence diagram illustrates the relationship between the respective classes involved in the creation and execution of a create command.
+
+![Create Command Sequence Diagram](./images/CreateCommand_Sequence.png)
+
+#### 1.6.2 Exit Command
+
+An Exit command can be used to exit the application.
+
+The following sequence diagram illustrates the relationship between the respective classes involved in the creation and execution of an exit command.
+
+![Exit Command Sequence Diagram](./images/ExitCommand_Sequence.png)
+
+#### 1.6.3 Help Command
+
+A Help command can be used to exit the application.
+
+The following sequence diagram illustrates the relationship between the respective classes involved in the creation and execution of a help command.
+
+![Help Command Sequence Diagram](./images/HelpCommand_Sequence.png)
+
+#### 1.6.4 Add Command
+
+An add command can be used to add a lesson to the timetable or add a module mapping to the user university list.
+
+The following sequence diagram illustrates the relationship between the respective classes involved in the creation and execution of an add command.
+
+![Add Command Sequence Diagram](./images/AddCommand_Sequence.png)
+
+#### 1.6.5 Delete Command
+
+A delete command can be used to delete a lesson from the timetable, delete a module mapping from the user university list or delete an entire user created university list.
+
+The following sequence diagram illustrates the relationship between the respective classes involved in the creation and execution of a delete command.
+
+![Delete Command Sequence Diagram](./images/DeleteCommand_Sequence.png)
```

Comment:
You may consider adding `\n` to very long texts in the diagram to avoid the long text for prolonging the the diagram. It would be more readable this way.